infra/token-transfer-server/src/lib/vesting.js

```diff
@@ -21,93 +21,123 @@ function momentizeGrant(grant) {
+  // with length of the number of remaining vesting events with that value
   const remainingVestingCount = vestingEventCount - cliffVestingCount
   const remainingVestingAmounts = Array(remainingVestingCount).fill(
     vestedPerEvent.integerValue(BigNumber.ROUND_FLOOR)
   )
 
+  // Complete vesting array
   const vestingEvents = [cliffVestAmount, ...remainingVestingAmounts]
+
+  // Add an rounding errors to the last vesting event
   const roundingError = BigNumber(grant.amount).minus(
     vestingEvents.reduce((a, b) => a.plus(b), BigNumber(0))
   )
   vestingEvents[vestingEvents.length - 1] = vestingEvents[
     vestingEvents.length - 1
   ].plus(roundingError)
 
-  return vestingEvents
-}
-
-/* Returns an array of vesting objects that include a datetime and status
- * associated with each vesting event.
- */
-function vestingSchedule(grantObj /* start = null, end = null */) {
-  const now = grantObj.now || moment()
-  const grant = momentizeGrant(grantObj)
-
-  // TODO implement filtering on start and end
-  /*
-  let bottomThreshold, topThreshold
-  if (start) {
-    bottomThreshold = start.diff(grant.cliff, grant.interval)
-  }
-  if (end) {
-    topThreshold = end.diff(grant.cliff, grant.interval)
-  }
-  */
-
-  return vestingEvents(grant).map((currentVestingEvent, index) => {
+  const events = vestingEvents.map((currentVestingEvent, index) => {
     const vestingEventDate = grant.cliff.clone()
     if (index > 0) {
-      vestingEventDate.add(index, grant.interval)
+      vestingEventDate.add(index, 'months')
     }
     return {
       amount: currentVestingEvent,
       date: vestingEventDate,
-      vested: vestingEventDate < now
+      vested: vestingEventDate <= now
     }
   })
+  return events
 }
 
-/* Returns the number of tokens vested by a grant.
- *
- */
-function vestedAmount(grantObj) {
-  const now = grantObj.now || moment()
+function investorVestingSchedule(grantObj) {
   const grant = momentizeGrant(grantObj)
+  const now = moment.utc()
+
+  const vestingSchedule = []
 
-  if (now < grant.cliff) {
-    return BigNumber(0)
-  } else if (now > grant.end) {
-    return BigNumber(grant.amount)
-  } else {
-    // Number of vesting events that have occurred since the cliff determines
-    // the index of the array for calculating events that have already vested
-    const threshold = now.diff(grant.cliff, grant.interval)
-    // Buld array of already vested amounts
-    const vested = vestingEvents(grant).slice(0, threshold + 1)
-    return vested.length ? Math.round(BigNumber.sum(...vested)) : BigNumber(0)
+  // Calculate initial vest percentage granted on grant start date
+  const initialVestPercentage = 6
+  // Time after which regular quarterly vesting begins
+  const quarterlyVestDelayMonths = 4
+  const firstQuarterlyVestDate = grant.start
+    .clone()
+    .add(quarterlyVestDelayMonths, 'months')
+  const quarterlyVestingPercentage = (100 - initialVestPercentage) / 8
```

Comment:
Regarding the calculation of quarterlyVestingPercentage  and quarterlyVestingAmount, since I think we had decided to not support decimal, should this be rounded down so that it is an integer? Then the last vesting amount would need to be adjusted so that the total vested amounts equals the grant amount

Reply:
Right you are sir, I need to add another round of tests for employee/investor vesting that doesn't round nicely like the values I've chose